src/digraph.rs

```diff
@@ -2889,6 +2889,48 @@ impl PyDiGraph {
         }
     }
 
+    /// Filters a graph's nodes by some criteria conditioned on a node's data payload and returns those nodes' indices
+    ///
+    /// :param filter_function: Function with which to filter nodes
```

Comment:
I think this needs a little more detail on how this is used. I think we should include what the format is for the filter function (what it gets passed and the required return type). Maybe having an example in the docs here would be useful too since it might be unclear even with an English description of the behavior.

This comment applies to all the functions being added in both graph and digraph.

Reply:
I think the example goes a long way for clarity thanks for adding that. But I still think we should explicitly document the inputs and required output from `filter_function` so that it's clear to users that it will be passed a node weight and is required to return a boolean value (this applies to all the docstrings for the other 3 methods too).